### PR TITLE
Add Trimmomatic process

### PR DESCRIPTION
**Pull request by JenkoB**
TODO:
- [x] add Trimmomatic tool to Docker image

### CODE REVIEW

**Comment by mstajdohar** [comment]
Rebase to the latest master.

File: resolwe_bio/processes/reads_processing/trimmomatic.yml [hunk 1]
```diff
@@ -0,0 +1,415 @@
+---
+
+- slug: trimmomatic-single
+  name: trimmomatic (single-end)
+  data_name: '{{ reads.fastq.0.file|default:"?" }}'
+  version: 0.0.1
+  type: data:reads:fastq:single:trimmomatic
+  category: analyses
+  flow_collection: sample
```
Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: This line is not required and will also be removed from other reads-filtering processors.

File: resolwe_bio/processes/reads_processing/trimmomatic.yml [hunk 2]
```diff
@@ -0,0 +1,415 @@
+---
+
+- slug: trimmomatic-single
+  name: trimmomatic (single-end)
```
Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: Capitalize first letter so that the label will be consistent with other tools in the UI.

File: resolwe_bio/processes/reads_processing/trimmomatic.yml [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+
+      re-save-file-list fastq ${NAME}_filtered.fastq.gz
+      re-save-file stats ${NAME}_stats.txt
+
+      re-save-file-list fastqc_archive ${NAME}_fastqc.zip
+      re-save-list fastqc_url $FASTQC_URL
+
+
+- slug: trimmomatic-paired
+  name: trimmomatic (paired-end)
```
Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: See above

File: resolwe_bio/processes/reads_processing/trimmomatic.yml [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+  data_name: '{{ reads.fastq.0.file|default:"?" }}'
+  version: 0.0.1
+  type: data:reads:fastq:paired:trimmomatic
+  category: analyses
+  flow_collection: sample
```
Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: See above

File: resolwe_bio/processes/reads_processing/trimmomatic.yml [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+    - name: fastqc_url2
+      label: Quality control with FastQC (Downstream)
+      type: list:basic:url:view
+    - name: fastqc_archive
+      label: Download FastQC archive (Upstream)
+      type: list:basic:file
+    - name: fastqc_archive2
+      label: Download FastQC archive (Downstream)
+      type: list:basic:file
+
```
Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: Remove empty line

**Comment by jkokosar** [comment]
@JenkoB Some committed files are not used in tests. Remove what is not necessary.

**Comment by JenkoB** [comment]
fixed

File: resolwe_bio/processes/reads_processing/trimmomatic.yml (hunk 1 above)
Comment (mstajdohar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: Why not required? Shouldn't the filtered reads be part of the sample.

Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: All filtered reads are derived from the "data:reads:fastq" objects which itself are already part of the sample - filtered reads objects should thus be added to the already existing sample object. Does "flow_collection: sample" needs to be added in such cases?

Comment (dblenkus) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: Yes,, this it required if you want to include Data object into Sample. It will be added to existing Sample if one exist and new one will be created otherwise.

Comment (jkokosar) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: OK, thank you both for the clarification. @JenkoB please revert this change back to the way it was before.

Comment (JenkoB) [inline resolwe_bio/processes/reads_processing/trimmomatic.yml]: Fixed back to first version.

File: resolwe_bio/tests/processes/test_reads_filtering.py [hunk 6]
```diff
@@ -105,3 +105,34 @@ def test_sortmerna_paired(self):
...
+
+    def test_trimmomatic_single(self):
+        reads = self.prepare_reads()
+        inputs = {'reads': reads.pk,
+                  'trailing': 3,
+                  'crop': 5}
+        filtered_reads = self.run_processor('trimmomatic-single', inputs)
+
+        self.assertFiles(filtered_reads, 'fastq', ['filtered_reads_trimmomatic_single.fastq.gz'], compression='gzip')
+        self.assertFields(filtered_reads, "fastqc_url", [{'url': 'fastqc/reads_fastqc/fastqc_report.html',
```
Comment (tjanez) [inline resolwe_bio/tests/processes/test_reads_filtering.py]: Just a cosmetic comment, please replace double quotes with single quotes. We use double quotes for strings meant for human consumption and single quotes for everything else.

**Comment by tjanez** [comment]
Please, rebase to master.

Otherwise, LGTM.
